unyt/dimensions.py

```diff
@@ -318,9 +321,30 @@ def returns(r_unit):
     Traceback (most recent call last):
     ...
     TypeError: result '6 m' does not match (length)/(time)
-
+    >>> @returns(length, length/time**2)
+    ... def f(a, v):
+    ...     return a * v, v / a
+    ...
+    >>> res = f(a= 2 * u.s, v = 3 * u.m/u.s)
+    >>> print(*res)
+    6 m 1.5 m/s**2
     """
 
+    # Convert deprecated arguments into current ones where possible.
+    if r_unit is not None:
+        if len(r_units) > 0:
+            raise ValueError(
+                "Cannot specify `r_unit` and other return values simultaneously"
+            )
+        else:
+            warnings.warn(
+                "Use of the @returns(r_unit=...) syntax is deprecated. "
+                "Please use @returns(...) instead.",
+                category=DeprecationWarning,
+                stacklevel=2,
+            )
```

Comment:
we have a thin wrapper function for this in `unyt._deprecation` that I advise you use instead of a raw `DeprecationWarning`. I should have remembered and told you that earlier, sorry !

Reply:
Done. I'm not sure which version of unyt this will be deprecated in, so I went with 3.0 to match some other warnings I saw. Please let me know if you'd like that changed (or feel free to change it without me).

Reply:
3.0 is indeed the next planned version, you're all good